apps/web/src/api/notification-templates.ts

```diff
@@ -1,11 +1,19 @@
-import { ICreateNotificationTemplateDto, INotificationTemplate, IGroupedBlueprint } from '@novu/shared';
+import {
+  ICreateNotificationTemplateDto,
+  INotificationTemplate,
+  IGroupedBlueprint,
+  IPaginationWithQueryParams,
+} from '@novu/shared';
 
 import { api } from './api.client';
 import { BLUEPRINTS_API_URL } from '../config';
 
-export function getNotificationsList(page = 0, limit = 10) {
-  return api.getFullResponse(`/v1/notification-templates`, { page, limit });
+export function getNotificationsList({ page = 0, limit = 10, query }: IPaginationWithQueryParams) {
+  const params = { page, limit, ...(query && { query }) };
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** Is there another way we could write the query portion?‏

Reply:
if statement or ternary